lib/poller.test.js

```diff
@@ -696,18 +701,84 @@ describe('Poller', () => {
       })).to.equal(true)
     })
 
+    it('doesn\'t update a secret if it hasn\'t changed', async () => {
```

Comment:
This test can't really check whether it has been updated, patched or nothing happened. Let's create a test to explicitly test `_equalSecretData()` instead to ensure it returns the correct value for the given arguments.

Reply:
Added the assertions to not have put or post called